src/pages/nft/claim/index.vue

```diff
@@ -449,40 +649,85 @@ export default {
+    loginContentArray() {
+      return [
+        {
+          title: this.$t('nft_claim_login_title_collection'),
+          content: this.$t('nft_claim_login_content_collection'),
+          icon: 'IconFolder',
+        },
+        {
+          title: this.$t('nft_claim_login_title_identity'),
+          content: this.$t('nft_claim_login_content_identity'),
+          icon: 'IconIdentity',
+        },
+        {
+          title: this.$t('nft_claim_login_title_community'),
+          content: this.$t('nft_claim_login_content_community'),
+          icon: 'IconCommunity',
+        },
+      ];
     },
-    errorMessage() {
-      if (
-        this.claimingAddressInput &&
-        !LIKE_ADDRESS_REGEX.test(this.claimingAddressInput)
-      ) {
-        return this.$t('nft_claim_enter_address_errorMessage');
-      }
-      return '';
+    loginUserDisplayName() {
+      return this.getLikerInfo?.displayName;
+    },
+    shouldShowStartReadingButton() {
+      return (
+        this.state === NFT_CLAIM_STATE.CLAIMED &&
+        this.isAutoDeliver &&
+        this.nftId
+      );
+    },
+    formatDownloadLinks() {
+      return this.classIds?.map(id => ({
+        canViewNFTBookBeforeClaim: this.getCanViewNFTBookBeforeClaimByClassId(
+          id
+        ),
+        url:
+          this.getIscnData(id)?.contentMetadata.url ||
+          this.getNFTClassMetadataById(id)?.external_url,
+        id,
+        contentUrls: this.getIscnData(id)?.contentMetadata.sameAs,
+        iscnUrl: this.getIscnData(id)?.contentMetadata.url,
+        isNftBook: this.checkNftIsNFTBook(id),
+        isContentViewable: true,
+        isDownloadable: !this.getIsHideNFTBookDownload(id),
+      }));
+    },
+    shouldDisplayDownloadOptions() {
+      return (
+        this.state === NFT_CLAIM_STATE.WELCOME &&
+        this.canViewContentDirectly &&
+        !this.isPhysicalOnly
+      );
     },
   },
   watch: {
-    walletEmail() {
-      this.claimingFreeEmail = this.walletEmail;
+    walletEmail(newValue) {
+      this.claimingFreeEmail = newValue;
     },
-    async getAddress() {
-      this.claimingAddressInput = this.getAddress;
-      if (this.isFreePurchase) this.claimingAddress = this.getAddress;
-      if (this.claimingAddress) {
-        await this.fetchRecommendInfo();
+    getAddress(newValue) {
+      if (this.isFreePurchase) this.claimingAddress = newValue;
+      if (!newValue && !this.loginAddress && !(this.status === 'completed')) {
+        this.claimingAddress = '';
+        this.state = NFT_CLAIM_STATE.LOGIN;
       }
     },
-    async loginAddress() {
-      this.claimingAddressInput = this.loginAddress;
-      this.claimingAddress = this.loginAddress;
-      if (this.claimingAddress) {
-        await this.fetchRecommendInfo();
+    loginAddress(newValue) {
+      this.claimingAddress = newValue;
+      if (!this.getAddress && !newValue && !(this.status === 'completed')) {
+        this.claimingAddress = '';
```

Comment:
seems redundant, the `!newValue` condition means `this.claimingAddress` and `newValue` is likely already empty in line 717

Reply:
[🔥 Remove used code](https://github.com/likecoin/liker-land/pull/1588/commits/857fc127de57cb34b7165f4c60c42517546f885f)